data-prepper-plugin-framework/src/main/java/org/opensearch/dataprepper/plugin/PluginBeanFactoryProvider.java

```diff
@@ -58,7 +60,20 @@ GenericApplicationContext getCoreApplicationContext() {
      * @return BeanFactory A BeanFactory that inherits from {@link PluginBeanFactoryProvider#sharedPluginApplicationContext}
      */
     public BeanFactory get() {
```

Comment:
I don't think we need the `get()` method anymore. Maybe remove the `Provider<BeanFactory>` interface and then remove `get`.

Reply:
Removed it.